src/organisms/Address/Address.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { ChangeEvent, Component, FormEvent } from 'react';
+
+import styled from '_styled-components';
+import { Button, Identicon } from 'atoms';
+import { Copyable } from 'molecules';
+import { borderRadius, scale } from 'Theme';
+import Typography from 'Typography';
+
+const Flex = styled.div`
+  display: flex;
+`;
+
+const Content = styled.div`
+  padding-left: ${scale(0)};
+`;
+
+const Title = styled(Typography)<{ clickable: boolean }>`
+  display: inline;
+  font-size: ${scale(1)};
+  ${props => props.clickable && `cursor: pointer;`};
+`;
+
+Title.defaultProps = { as: 'div' };
+
+const TitleInput = styled.input`
+  background: ${props => props.theme.background};
+  border: 0.125em solid ${props => props.theme.controlBorder};
+  border-radius: ${borderRadius};
+  color: ${props => props.theme.text};
+  font: ${scale(1)} Lato, sans-serif;
+  padding: 0;
+`;
+
+const ColoredIconButton = styled(Button)`
+  color: #b5bfc7;
+`;
+
+const SubmitButton = styled(ColoredIconButton)`
+  color: #1eb8e7;
+  left: -${scale(2)};
+  position: relative;
+`;
+
+SubmitButton.defaultProps = { type: 'submit', icon: 'check' };
+
+interface Props {
+  address: string;
+  title: string;
+  onSubmit?(title: string): void;
+  truncate?(text: string): string;
+}
+
+interface State {
+  editing: boolean;
+  title: string;
+}
+
+export class Address extends Component<Props, State> {
+  public constructor(props: Props) {
```

Comment:
Kinda curious about using the constructor vs getDerivedStateFromProps

Reply:
Maybe I should actually use getDerivedStateFromProps in case a parent wants to update the title, but that would mean that the title may change the current input field while being edited by the user. With the current implementation, you lose the parent's update if the user is editing the title. What do you think is better?